tests/validator_test.go

```diff
@@ -426,10 +426,13 @@ var _ = Describe("Template validator webhooks", func() {
 			waitForDeletion(client.ObjectKeyFromObject(vm), vm)
 		}
 		if template != nil {
-			err := apiClient.Delete(ctx, template)
-			if !errors.IsNotFound(err) {
-				Expect(err).ToNot(HaveOccurred(), "Failed to delete Template")
-			}
+			Eventually(func() bool {
+				err := apiClient.Delete(ctx, template)
+				if err != nil && !errors.IsNotFound(err) {
+					fmt.Println("template wasn't deleted: " + err.Error())
+				}
+				return errors.IsNotFound(err)
+			}, shortTimeout).Should(BeTrue(), "Template should be deleted")
```

Comment:
This will print the log line on every iteration. Can you use this instead:
```go
Eventually(func(g Gomega) {
   if err := apiClient.Delete(ctx, template); err != nil {
      g.Expect(errors.ReasonForError(err)).To(Equal(metav1.StatusReasonNotFound))
   }
}, shortTimeout, time.Second).Should(Succeed(), "Template should be deleted")
```

It is a special form of the `Eventually`, but it is useful in this case. Hopefully the error message will be meaningful.
https://pkg.go.dev/github.com/onsi/gomega#Eventually